### PR TITLE
Feature/module 4

### PR DESCRIPTION
**Pull request by andrei-shelenhouski**

### CODE REVIEW

File: src/services/group.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  public async updateGroup(id: string, groupUpdates: IGroup): Promise<IGroup | null | undefined> {
+    const [numberOfItems, updatedItems]: [number, IGroup[]] = await GroupModel.update(groupUpdates, {
+      where: {
+        id,
+      },
+    });
+
+    return numberOfItems ? updatedItems.shift() : null;
```
Comment (kuzanatoliy) [inline src/services/group.service.ts]: Looks as you can use destruction for getting first element

File: src/services/group.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    return deletedRowsCount;
+  }
+
+  public async addUsersToGroup(id: string, userIds: string[]): Promise<void> {
+    try {
+      await sequelize.transaction(async t => {
+        return Promise.all(
+          (userIds || []).map(async userId => {
+            return UserGroupModel.create(
```
Comment (kuzanatoliy) [inline src/services/group.service.ts]: Please think about using bulkCreate in this and other cases

File: src/services/group.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        id,
+      },
+    });
+
+    return deletedRowsCount;
+  }
+
+  public async addUsersToGroup(id: string, userIds: string[]): Promise<void> {
+    try {
+      await sequelize.transaction(async t => {
```
Comment (kuzanatoliy) [inline src/services/group.service.ts:48]: Please use readable naming

File: src/services/group.service.ts (hunk 1 above)
Comment (andrei-shelenhouski) [reply src/services/group.service.ts]: refactored

File: src/services/group.service.ts (hunk 3 above)
Comment (andrei-shelenhouski) [reply src/services/group.service.ts:48]: t - means transaction (according to documentation). but I will keep in mind

File: src/services/group.service.ts (hunk 2 above)
Comment (andrei-shelenhouski) [reply src/services/group.service.ts]: Done

**Pull request merged**